### DISCUSSION TITLE
Unknown User in sentry issue details

### DISCUSSION THREAD

**Discussion by ahmedspiir** [getsentry/sentry-dotnet#888] [Q&A]
Hi 
I'm migrating from https://github.com/getsentry/raven-csharp to the new SDK, before i was able to see the user information and In the unresolved issues list on Sentry ,i was able to see the users count, now and since i migrated it's always zero.
I'm using `Sentry.Serilog`  sentry.dotnet.serilog Version | 1.1.2
My application is a MS Service fabric service, running on .net 4.6.1
and here is a configurations
```csharp
  loggerConfig = loggerConfig.WriteTo.Sentry(o =>
               {
                   o.Environment = environment;
                   o.Release = FabricRuntime.GetActivationContext().GetServiceManifestVersion();
                   o.Dsn = new Dsn(settings.SentryDsn);
                   o.MinimumEventLevel = LogEventLevel.Error;
               });
                SentrySdk.ConfigureScope(scope => { scope.User = GetUser(); });

// GetUser returns Sentry.Protocol.User and it contains only the username 
private static User GetUser(){}
``` 
<img width="1077" alt="Screenshot 2019-04-16 at 16 13 58" src="https://user-images.githubusercontent.com/43743573/56217045-d5b54900-6062-11e9-9124-867d6840dfb1.png">
<img width="217" alt="Screenshot 2019-04-16 at 16 15 51" src="https://user-images.githubusercontent.com/43743573/56217113-f1b8ea80-6062-11e9-9793-0f8c466eeaf6.png">

**Comment by bruno-garcia** [comment]
@ahmedspiir Do you build the logging factory before calling `ConfigureScope`?
It seems like you're only making the `loggerConfig` at that point, then calling `ConfigureScope` before then SDK is actually initialized which is a no-op.

Could you make sure the flow is:

```csharp
var loggerConfig = WriteTo.Sentry(...);
var logger = loggerConfig.CreateLogger(); // Will invoke the configuration callback defined above and init the SDK.
// Now SDK is initialized, scope can be modified:
SentrySdk.ConfigureScope(scope => { scope.User = GetUser(); });
```

**Comment by ahmedspiir** [comment]
@bruno-garcia Thanks for the help, i'll try it and tell you how it went.

**Comment by bruno-garcia** [comment]
Just realized that your environment has no `Sentry.AspNetCore` integration so reopening this issue.

Please let us know what result of the test I suggested.

**Comment by bruno-garcia** [comment]
@ahmedspiir any updates on the repro?

**Comment by ahmedspiir** [comment]
@bruno-garcia we are going to test the change this week, and once it's out i'll post you back

**Comment by kanadaj** [comment]
The SendDefaultPii flag is not set in the options; maybe that's the issue?

**Comment by bruno-garcia** [comment]
@kanadaj that makes sense!

**Comment by kanadaj** [comment]
@bruno-garcia Maybe consider making a compulsory true/false flag in the setup functions since this seems like a common issue, or add to FAQ

**Comment by ahmedspiir** [comment]
@bruno-garcia @kanadaj I have applied SendDefaultPii and upgraded to the latest version which is 1.2.0 of "sentry.dotnet.serilog".
But I still get the users to count as Zero "0".

**Comment by bruno-garcia** [comment]
@ahmedspiir Could you please provide us with a small repro so we can debug this case?
